packages/core/src/external/commonwell/management/session.ts

```diff
@@ -0,0 +1,195 @@
+  private readonly codeChallenge: CodeChallenge;
+  private readonly headless: boolean;
+  private browser: Browser | undefined;
+  private context: BrowserContext | undefined;
+  private sessionBaseUrl: string;
+  private username: string;
+  private password: string;
+  private debug: typeof console.log;
+
+  constructor(params: SessionManagementConfig) {
+    this.cookieManager = params.cookieManager;
+    this.cwManagementApi = params.cwManagementApi;
+    this.codeChallenge = params.codeChallenge;
+    this.headless = params.headless == undefined ? true : params.headless;
+    this.sessionBaseUrl = this.cwManagementApi.getBaseUrl();
+    this.username = params.username;
+    this.password = params.password;
+    this.debug = params.debug ?? (() => {}); //eslint-disable-line @typescript-eslint/no-empty-function
+  }
+
+  async keepSessionActive(): Promise<void> {
+    const log = console.log;
+    try {
+      const isSessionValid = (await this.cwManagementApi.getMember()) != undefined;
+
+      if (isSessionValid) {
+        log(`Session is still valid, returning...`);
+        return;
+      }
+
+      log(`Session is invalid, generating new cookies...`);
+      const page = await this.login();
+
+      const cookies = await page.context().cookies();
+      const filteresCookies = cookies.filter(this.isEnabledCookie);
+      this.debug(`-----> cookiesString: ${cookiesToString(filteresCookies)}`);
+
+      log(`Got cookies, updating store...`);
+      await this.cookieManager.updateCookies(filteresCookies);
+
+      log(`Done.`);
+    } finally {
+      await this.logout();
+    }
+  }
+
+  /**
+   * Logs in on the CommonWell management portal.
+   *
+   * @returns the Playwright Page object pointing to CW's management portal Organization List page.
+   */
+  protected async login(): Promise<Page> {
+    const log = this.debug;
+    if (!this.browser || !this.browser.isConnected() || !this.context) {
+      this.browser = await firefox.launch({ headless: this.headless, slowMo: 50 });
+      this.context = await this.browser.newContext(devices["Desktop Firefox HiDPI"]);
+    }
+    const page = await this.context.newPage();
+
+    try {
+      log(`.1`);
+      await page.goto(this.sessionBaseUrl);
+      log(`.2`);
+      assert(await page.getByRole("link", { name: "Healthcare ID" }).isEnabled());
+      await page.getByRole("link", { name: "Healthcare ID" }).click();
+      log(`.3`);
+      assert(await page.getByTestId("username").isEditable());
+      await page.getByTestId("username").click();
+      await page.getByTestId("username").fill(this.username);
+      log(`.5`);
+      assert(await page.getByTestId("login-pwd").isEditable());
+      await page.getByTestId("login-pwd").click();
+      await page.getByTestId("login-pwd").fill(this.password);
+      log(`.7`);
+      assert(await page.getByRole("button", { name: "Continue" }).isEnabled());
+      await page.getByRole("button", { name: "Continue" }).click();
```

Comment:
nit: can put common strings in const so we don't have to update in multiple places if needing to add/change them

Reply:
Yeah, can defs improve a lot here. Maybe something for when we nee do update this b/c of some UI change?